### DISCUSSION TITLE
Why PR using CLI don't need to provide the correct acount name, only need the correct token

### DISCUSSION THREAD

**Discussion by HidayatLahabu** [community/community#62028] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

I tested git pull origin [branch], like this :
ubuntu@ubuntu-ct-01:/var/www/html/simblud/farmasi$ sudo git pull origin master
Username for 'https://github.com': a
Password for 'https://a@github.com': 
remote: Enumerating objects: 21, done.
remote: Counting objects: 100% (21/21), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 12 (delta 7), reused 12 (delta 7), pack-reused 0
Unpacking objects: 100% (12/12), 5.12 KiB | 77.00 KiB/s, done.
From https://github.com/xxxxxxx/simblud
 * branch              master     -> FETCH_HEAD
   95f60478..e17fb0e6  master     -> origin/master
Merge made by the 'ort' strategy.
 farmasi/app/Http/Controllers/Admin/Farmasi/LaporanResepController.php |  5 -----
 farmasi/app/Http/Controllers/Admin/Farmasi/PostingResepController.php | 19 -------------------
 farmasi/app/Http/Controllers/Admin/Persediaan/LaporanController.php   |  4 ++--
 3 files changed, 2 insertions(+), 26 deletions(-)

even that Username for 'https://github.com': a, is not my real account name, the pull request still can be execute
is anybody had tough the risk of this method ?

**Comment by IshaanAdarsh** [comment]
To avoid the risk of using your GitHub credentials directly in the terminal, you can set up an SSH key or generate a personal access token and use that for authentication instead.  (SSH key>>>)

**Reply by HidayatLahabu** [reply]
The use of a personal access token should provide the appropriate authorization to execute pull requests on GitHub. However, if your account name is not correct, you can still make a pull request by adding the correct personal access token. This poses a potentially risky situation because if someone gets hold of your personal access token, even if they don't know your account name, they can still make pull requests from your GitHub account.

This should not happen. GitHub should verify the compatibility between the used personal access token and the account name sent with the request. If there is a mismatch between the token and the account, the pull request should be rejected.

**Comment by mujeeb00** [comment, marked as answer]
The reason why the pull request using the CLI doesn't require the correct account name but only needs the correct token is due to the authentication mechanism used by GitHub when accessing repositories over HTTPS.

When you use the git pull command with an HTTPS URL (e.g., https://github.com/username/repo.git), Git will prompt you for a username and password for authentication. However, since GitHub introduced token-based authentication, using a personal access token has become the preferred method for authentication.

When you provide the token in place of the password during the authentication process, GitHub recognizes it as a valid authentication token associated with your account. The token serves as a proof of your identity and access permissions to the repository. As long as the token is valid and has the necessary permissions to perform the actions (e.g., pull requests), GitHub will allow the requested operations to be executed, even if the username provided is incorrect or doesn't match the actual account name.

The risk associated with using the token-based authentication method is relatively low, especially if you are keeping your token secure and not sharing it with others. However, it's essential to follow good security practices:

Keep Your Token Secure: Treat your personal access token as you would treat your password. Don't share it openly, and avoid storing it in plain text in scripts or config files.

Use Token Scopes Wisely: When generating a personal access token, only grant the necessary scopes and permissions required for the specific tasks you intend to perform. Minimize access to sensitive data or actions.

Rotate Tokens Regularly: Consider rotating your tokens periodically, especially if you suspect any potential security compromise.

Use SSH for Increased Security: Alternatively, you can use SSH for Git operations, which offers an additional layer of security and doesn't require a token or password for every interaction.

By following these best practices, you can use token-based authentication for CLI operations safely and efficiently. It is a widely accepted method in the GitHub community and is considered secure when used responsibly.
